### PR TITLE
Add grafana-o11y-ds-frontend workspace package

### PR DESCRIPTION
**Pull request by fabrizio-grafana**
This PR adds a new package, to be used as workspace in Grafana to share frontend code across data sources.

(Note that the branch has a legacy name, since we renamed the library in progress.)

### CODE REVIEW

File: packages/grafana-traces/package.json
Comment (jackw) [inline packages/grafana-traces/package.json:23]: You can delete this publishConfig, it's only used by yarn when packing for publish and this package is set as private.

File: packages/grafana-traces/package.json
Comment (jackw) [inline packages/grafana-traces/package.json:28]: This script can also be deleted.

File: packages/grafana-traces/package.json
Comment (jackw) [inline packages/grafana-traces/package.json:31]: Both of these can be deleted too.

File: packages/grafana-traces/src/IntervalInput/IntervalInput.test.tsx
Comment (jackw) [inline packages/grafana-traces/src/IntervalInput/IntervalInput.test.tsx:2]: I think you'll need to add these to the package.json as yarn won't allow resolving to deps in root package.json.

**Review by jackw — COMMENTED** [review]
Awesome stuff. Left a few comments for consideration.

File: packages/grafana-traces/package.json
Comment (aocenas) [inline packages/grafana-traces/package.json:3]: Is this a license change if this is code from core? Would again probably opt to keep the same license as core for now.

**Comment by aocenas** [comment]
Apart from that license comment, I assume there aren't changes to the code right, just moving things around? In that case though is this duplicating the code? Shouldn't this also clean up the code from core?

**Comment by jackw** [comment]
> In that case though is this duplicating the code? Shouldn't this also clean up the code from core?

I would vote this code is moved rather than copied in this PR and the imports in core are updated to use the workspace. 👍

File: packages/grafana-traces/src/utils.ts
Comment (aocenas) [reply packages/grafana-traces/src/utils.ts:40]: Don't remember writing this, seems like it may be something from jeager codebase but I would say this should actually return `[id: string]: { span: T; children: string[] }`.

The thing I assume is that it is creating a tree with children from a list of spans with parentIds. Those spans may not be in order of tree tarversal and so we can get child span before we get parent span. In that case we create a temporary span object to have a placeholder for the ID and fill in the span data later. We could recheck in the end to make sure and maybe just delete anything that is still `span: undefined` but even in that case I would say the type assertion will be necessary.

File: packages/grafana-traces/src/pyroscope/datasource.ts
Comment (joey-grafana) [inline packages/grafana-traces/src/pyroscope/datasource.ts:16]: We need this for `ProfileTypesCascader` but do we need all of the others?

File: packages/grafana-traces/package.json
Comment (joey-grafana) [inline packages/grafana-traces/package.json:3]: Are we sure this is correct? I ask because even though it's the same as core, our other packages use `Apache-2.0` so wondering why there's a difference.

**Review by joey-grafana — COMMENTED** [review]
Thanks for making these changes 👍 

Can we also remove `defaultGrafanaPyroscope` from `dataquery.gen.ts` - I would say basically anything that is not used in/needed by the package can be deleted.

Probably a pre-existing issue in Zipkin but just wanted to mention:
Screenshot 2024-01-17 at 10 32 33

File: packages/grafana-traces/package.json
Comment (aocenas) [reply packages/grafana-traces/package.json:3]: Yeah it was my suggestion. Imho if something is for outside consumption (like 3rd party plugins) then it should be `Apache-2.0` but if it's only for our own usage it should stay `AGPL`

Comment (jackw) [reply packages/grafana-traces/package.json:3]: > Yeah it was my suggestion. Imho if something is for outside consumption (like 3rd party plugins) then it should be `Apache-2.0` but if it's only for our own usage it should stay `AGPL`

Yep, if the code lives in this repository and is not published to NPM as a public package I also believe the `AGPL-3.0-only` licence is correct.

Comment (joey-grafana) [reply packages/grafana-traces/package.json:3]: Makes sense, thanks for clarifying.

File: packages/grafana-o11y-ds-frontend/package.json
Comment (jackw) [inline packages/grafana-o11y-ds-frontend/package.json:39]:
```suggestion
    "@types/systemjs": "6.13.5",
```

**Review by jackw — APPROVED** [review]
Changes look good to me from a frontend-ops perspective. 🚀 

I would seek additional approval from codeowners before merging.

File: scripts/check-breaking-changes.sh
Comment (jackw) [inline scripts/check-breaking-changes.sh:7]: Are you sure you wanna disable levitate on this package? Even though it's private it might help track down issues. 🤷

**Review by joey-grafana — APPROVED** [review]
Package updates LGTM 👍

File: scripts/check-breaking-changes.sh
Comment (jackw) [reply scripts/check-breaking-changes.sh:7]: Fair enough. This PR has highlighted that we're making a few assumptions in the levitate PR checks which causes failures. I'll try to get something on our board so we can remove this in the future.

**Pull request merged**
